src/voterData/voterFile/voterFile.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { CampaignWith } from 'src/campaigns/campaigns.types'
+import { GetVoterFileSchema } from './schemas/GetVoterFile.schema'
+import { CHANNEL_TO_TYPE_MAP } from './voterFile.types'
+import { typeToQuery } from './util/voterFile.util'
+import { VoterDataService } from '../voterData.service'
+import { User } from '@prisma/client'
+import { SlackService } from 'src/shared/services/slack.service'
+
+@Injectable()
+export class VoterFileService {
+  private readonly logger = new Logger(VoterFileService.name)
+
+  constructor(
+    private readonly voterDataService: VoterDataService,
+    private readonly slack: SlackService,
+  ) {}
+
+  async getCsv(
+    campaign: CampaignWith<'pathToVictory'>,
+    { type, countOnly, customFilters }: GetVoterFileSchema,
+  ) {
+    const resolvedType =
+      type === 'custom' && customFilters
+        ? CHANNEL_TO_TYPE_MAP[customFilters.channel]
+        : type
+
+    const countQuery = typeToQuery(resolvedType, campaign, customFilters, true)
+    this.logger.log('Count Query:', countQuery)
+    let withFixColumns = false
+    const sqlResponse = await this.voterDataService.query(countQuery)
+    const count = parseInt(sqlResponse.rows[0].count)
+    if (count === 0) {
+      // is it a string?.
+      withFixColumns = true
+    }
+    if (countOnly && count !== 0) {
+      return count
+    }
+    if (countOnly && count === 0) {
+      const countQuery = typeToQuery(
+        resolvedType,
+        campaign,
+        customFilters,
+        true,
+        true,
+      )
+      const sqlResponse = await this.voterDataService.query(countQuery)
+      const count = parseInt(sqlResponse.rows[0].count)
+      return count
+    }
+
+    this.logger.log('count', sqlResponse.rows[0].count)
```

Comment:
Maybe these should be `this.logger.debug` so that they won't be polluting production logs? 🤔